clang/test/Driver/sycl-offload.cpp

```diff
@@ -24,3 +24,11 @@
 // RUN:  %clang -### -fsycl %s 2>&1 | FileCheck -check-prefix=CHECK-OPTS %s
 // CHECK-OPTS: clang{{.*}} "-cc1" {{.*}} "-fsycl-is-device"
 // CHECK-OPTS: clang{{.*}} "-cc1" {{.*}} "-fsycl-is-host"
+
+/// Check that -fcoverage-mapping is disabled for device
+// RUN: %clang -### -fsycl -fprofile-instr-generate -fcoverage-mapping -target x86_64-unknown-linux-gnu -c %s 2>&1 \
+// RUN:  | FileCheck -check-prefix=CHECK_COVERAGE_MAPPING %s
+// CHECK_COVERAGE_MAPPING: clang{{.*}} "-cc1" "-triple" "spir64-unknown-unknown-sycldevice"{{.*}} "-fprofile-instrument=clang"
+// CHECK_COVERAGE_MAPPING-NOT: "-fcoverage-mapping"
+// CHECK_COVERAGE_MAPPING: clang{{.*}} "-cc1" "-triple" "x86_64-unknown-linux-gnu" {{.*}} "-fprofile-instrument=clang"
```

Comment:
I'd suggest adding `-fsycl-is-[device/host]` characteristics to the respective search strings. This would better illustrate the test case scenario and avoid potential interference from the integration header generation command.

Reply:
Funny you should say that. I was going to add it, but the locations in which host and device variants in the compilation strings are wildly different so it will take some additional massaging to make it look ok :)